python/lsst/ts/m2/tcp_client.py

```diff
@@ -124,6 +141,10 @@ async def connect(self, connect_retry_interval=1.0):
             except ConnectionRefusedError:
                 await asyncio.sleep(connect_retry_interval)
 
+                retry_times += 1
+                if retry_times >= retry_times_max:
+                    raise asyncio.TimeoutError("Connection timeout.")
```

Comment:
Not sure I understand this part. Why raise `TimeoutError` when the error is `ConnectionRefusedError`?  Isn't it better to just re-raise the exception? it might have some useful information no?

Reply:
This is to make sure I can try to do the connection request multiple times in a time period. If the connection can not be constructed, the `TimeoutError` will be raised. The `ConnectionRefusedError` is from `asyncio.open_connection()`, which is a non-block call and will raise the error immediately if the connection can not be constructed.